pyomo/contrib/mindtpy/config_options.py

```diff
@@ -203,6 +203,11 @@ def _get_MindtPy_config():
                     'and minimize the sum of the slack variables (Augmented Penalty).',
         domain=bool
     ))
+    CONFIG.declare('partition_obj_nonlinear_terms', ConfigValue(
+        default=True,
```

Comment:
Having this option on by default is supported by benchmarks? Is this the case with SHOT?

Reply:
We can add this option in the benchmark code. By default, SHOT will automatically partition obj nonlinear terms if the nonlinear terms are convex.

Reply:
We can add this option in the benchmark code. SHOT will automatically partition obj nonlinear terms if the nonlinear terms are convex by default.

Reply:
It works well for convex cases, but since we do not check for convexity, I do not know whether this partition would work well with nonconvex and augmented penalty for example.
Let's benchmark it and decide based on it.

Reply:
No problem.

Reply:
There is also the option to use SUSPECT (if present) to do convexity detection.

Reply:
Good point @jsiirola. Suspect can do FBBT as well.